x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/create/helpers.test.ts

```diff
@@ -503,19 +502,164 @@ describe('helpers', () => {
+        actions: [],
+        enabled: false,
+        meta: {
+          throttle: 'no_actions',
+          kibanaSiemAppUrl: 'http://localhost:5601/app/siem',
+        },
+        throttle: null,
+      };
+
+      expect(result).toEqual(expected);
+    });
+
+    test('returns proper throttle value for no_actions', () => {
+      const mockStepData = {
+        ...mockData,
+        throttle: 'no_actions',
+      };
+      const result: ActionsStepRuleJson = formatActionsStepData(mockStepData);
+      const expected = {
+        actions: [],
+        enabled: false,
+        meta: {
+          throttle: mockStepData.throttle,
+          kibanaSiemAppUrl: mockStepData.kibanaSiemAppUrl,
+        },
+        throttle: null,
+      };
+
+      expect(result).toEqual(expected);
+    });
+
+    test('returns proper throttle value for rule', () => {
+      const mockStepData = {
+        ...mockData,
+        throttle: 'rule',
+        actions: [
+          {
+            group: 'default',
+            id: 'id',
+            actionTypeId: 'actionTypeId',
+            params: {},
+          },
+        ],
+      };
+      const result: ActionsStepRuleJson = formatActionsStepData(mockStepData);
+      const expected = {
+        actions: [
+          {
+            group: mockStepData.actions[0].group,
+            id: mockStepData.actions[0].id,
+            action_type_id: mockStepData.actions[0].actionTypeId,
+            params: mockStepData.actions[0].params,
+          },
+        ],
+        enabled: false,
+        meta: {
+          throttle: mockStepData.throttle,
+          kibanaSiemAppUrl: mockStepData.kibanaSiemAppUrl,
+        },
+        throttle: null,
+      };
+
+      expect(result).toEqual(expected);
+    });
+
+    test('returns proper throttle value for interval', () => {
+      const mockStepData = {
+        ...mockData,
+        throttle: '1d',
+        actions: [
+          {
+            group: 'default',
+            id: 'id',
+            actionTypeId: 'actionTypeId',
+            params: {},
+          },
+        ],
+      };
+      const result: ActionsStepRuleJson = formatActionsStepData(mockStepData);
+      const expected = {
+        actions: [
+          {
+            group: mockStepData.actions[0].group,
+            id: mockStepData.actions[0].id,
+            action_type_id: mockStepData.actions[0].actionTypeId,
+            params: mockStepData.actions[0].params,
+          },
+        ],
+        enabled: false,
+        meta: {
+          throttle: mockStepData.throttle,
+          kibanaSiemAppUrl: mockStepData.kibanaSiemAppUrl,
+        },
+        throttle: mockStepData.throttle,
+      };
+
+      expect(result).toEqual(expected);
+    });
+
+    test('returns actions with action_type_id', () => {
+      const mockAction = {
+        group: 'default',
+        id: '99403909-ca9b-49ba-9d7a-7e5320e68d05',
+        params: { message: 'ML Rule generated {{state.signalsCount}} signals' },
```

Comment:
I believe this is the old count field as I was not seeing count come through when testing with the default message.

```suggestion
        params: { message: 'ML Rule generated {{state.signals_count}} signals' },
```